### PR TITLE
Changed logic of halt to be annotation based.

### PR DESCRIPTION
**Pull request by fran2810**
Removed enabled from spec and changed halt to be based on annotation.

### CODE REVIEW

File: api/v1alpha1/capp_types.go [hunk 1]
```diff
@@ -138,13 +134,13 @@ type RevisionInfo struct {
...
-	// +kubebuilder:default:=false
+type StateStatus struct {
+	// State is actual enabled state of the capp
+	// +kubebuilder:default:="running"
 	// +optional
-	IsEnabled bool `json:"isEnabled,omitempty"`
+	State string `json:"state,omitempty"`
 
-	// LastEnabledChange the last time the enabled state of capp changed
+	// LastChange the last time the enabled state of capp changed
```
Comment (mzeevi) [inline api/v1alpha1/capp_types.go:143]: // LastChange is the last time the enabled state of Capp changed

File: internals/resource-managers/knative_service.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 
 const (
 	danaAnnotationsPrefix = "rcs.dana.io"
+	capphaltState         = "halted"
```
Comment (mzeevi) [inline internals/resource-managers/knative_service.go:25]: Change to cappHaltState

File: internals/utils/common.go [hunk 3]
```diff
@@ -8,6 +8,10 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+const (
+	danaHaltAnnotation = "capp.dana.io/state"
```
Comment (mzeevi) [inline internals/utils/common.go:12]: Rename to cappHaltAnnotation

**Review by mzeevi — CHANGES_REQUESTED** [review]
Change some minor naming

File: api/v1alpha1/capp_types.go (hunk 1 above)
Comment (fran2810) [reply api/v1alpha1/capp_types.go:143]: i also removed the enabled

**Review by mzeevi — APPROVED** [review]

**Pull request merged**